### PR TITLE
Split up crates to facilitate backend switching

### PR DESCRIPTION
**Pull request by Manishearth**
This gets rid of the gst feature, instead using target-switching to select the backend.

The servo-media crate is a thin wrapper that does all the backend switching. The GStreamer backend is 
now in the servo-media-gstreamer crate, and all of the audio stuff is in the servo-media-audio crate.

You use the crate the same way (one small difference, in accordance to rust style you will have to import the crate as `servo-media` in Cargo.toml), and it automatically picks the backend based on the platform.

This makes it pretty easy to swap backends as well; adding a new backend can be done with a cargo feature, and once we fix the android stuff we can make this unconditional.

r? @ferjm

### CODE REVIEW

File: servo-media-audio/src/node.rs.orig [hunk 1]
```diff
@@ -0,0 +1,225 @@
+<<<<<<< HEAD:servo-media-audio/src/node.rs
```
Comment (CYBAI) [inline servo-media-audio/src/node.rs.orig]: Maybe this one is commited accidentally?

**Comment by Manishearth** [comment]
I had to git merge just because rebasing was throwing up way too many merge conflicts.

File: servo-media-audio/src/node.rs.orig (hunk 1 above)
Comment (Manishearth) [reply servo-media-audio/src/node.rs.orig]: fixed

**Comment by Manishearth** [comment]
I'd kinda mentioned this last week but I didn't really explain it well 😄

**Pull request merged**